sql2.txt

```diff
@@ -2,51 +2,84 @@
 SUM(Quantity) AS TotalCount .  Order by product name.
 You will need to group by product ID.  Paste your SQL statement below.
+SELECT Products.ProductName, SUM(OrderDetails.Quantity) AS TotalCount
+FROM Products
+    JOIN OrderDetails ON Products.ProductID = OrderDetails.ProductID
+    GROUP BY Products.ProductID, Products.ProductName
+    ORDER BY Products.ProductName ASC;
 
 
 2. Retrieve the customer name and OrderDetailsID for all the OrderDetails for each customer,
 ordered by customer name. Paste your SQL statement below. You will have to join the Customers
 table with the Orders table, and the Orders table with the OrderDetails table, 
 because the Orders table is the one that ties the Customer table to the OrderDetails table.
-
+SELECT Customers.CustomerName, OrderDetails.OrderDetailID
+FROM OrderDetails
+    JOIN Customers on Orders.CustomerID = Customers.CustomerID
+    JOIN Orders on OrderDetails.OrderID = Orders.OrderID
+    GROUP BY Customers.CustomerName, OrderDetails.OrderDetailID 
+    ORDER BY Customers.CustomerName ASC;
 
 3. Extend the previous query to join the Products table.  You are now joining the Customers, Orders,
 OrderDetails, and Products tables.  The result of your query should include the
 customer name, the product name, and the total quantity of that product ordered.  Note that a customer
 may have ordered the same product in several different orders.  So the total quantity is
 SUM(Quantity) AS TotalQuantity and you must GROUP BY CustomerID, ProductID.  Paste your
 SQL statement below.
-
-
+SELECT Customers.CustomerName, Products.ProductName, SUM(OrderDetails.Quantity) AS TotalQuantity
+FROM Customers
+    JOIN Orders ON Customers.CustomerID = Orders.CustomerID
+    JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID
+    JOIN Products ON OrderDetails.ProductID = Products.ProductID
+GROUP BY Customers.CustomerID, Customers.CustomerName, Products.ProductID, Products.ProductName;
 
 4. Extend the previous query to give the total price as well as the customer name, product name, 
 and total quantity.  Here you will have
 (SUM(Quantity) * Price) AS TotalPrice.  Paste your SQL statement below.
-
+SELECT Customers.CustomerName, Products.ProductName, (SUM(OrderDetails.Quantity) * Price) AS TotalPrice
+FROM Customers
+    JOIN Orders ON Customers.CustomerID = Orders.CustomerID
+    JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID
+    JOIN Products ON OrderDetails.ProductID = Products.ProductID
+GROUP BY Customers.CustomerID, Customers.CustomerName, Products.ProductID, Products.ProductName;
 
 5. Add a new Order to the Orders table.  You will have to specify the ID of an existing customer
 to satisfy the foreign key constraint.  You will also have to specify the ID of an existing
 employee.  Paste your SQL statement below.
-
+INSERT INTO Orders(OrderID, CustomerID, EmployeeID, OrderDate, ShipperID)
+VALUES (10247, 90, 5, "6/25/2024", 3);
```

Comment:
I see that the lowest OrderID in the starter data is 10248, so this works, but since OrderID is an auto-incrementing primary key, we usually just let the database take care of setting the OrderID itself, like so:

```suggestion
INSERT INTO Orders(CustomerID, EmployeeID, OrderDate, ShipperID)
VALUES (90, 5, "6/25/2024", 3);
```